### PR TITLE
Docs: changes from 8.4 merge iteration 4

### PR DESCRIPTION
**Pull request by litzell**
Doc changes from PR 3492 are incorporated into gpdb 6 DITA docs. Starting with changes to create_type.sgml. Changes to previous files are in another PR.

### CODE REVIEW

**Comment by litzell** [comment]
Changes are staged here: http://docs-litzell-gpdb6.cfapps.io/600/common/gpdb-features.html

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml [hunk 1]
```diff
@@ -667,28 +669,36 @@ weather_reports ORDER BY location, time DESC;</codeblock><p>retrieves
           gains. However, when the number of distinct values is close to the total number of rows,
           the transformation might result in the generation of a multi-level grouping plan. In this
           case, there is an expected performance degradation because of the overhead introduced by
-          the lower aggregation level.</p></sectiondiv><sectiondiv id="section16"><b>The LIMIT
-          Clause</b><p>The <codeph>LIMIT</codeph> clause consists of two independent
-          sub-clauses:</p><codeblock>LIMIT {<varname>count</varname> | ALL}
+          the lower aggregation level.</p></sectiondiv><b>The LIMIT Clause</b><p>The
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: The LIMIT Clause should be in a <sectiondiv>

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml [hunk 2]
```diff
@@ -667,28 +669,36 @@ weather_reports ORDER BY location, time DESC;</codeblock><p>retrieves
...
-          unless <codeph>ORDER BY</codeph> is used to constrain the
-        order.</p></sectiondiv><sectiondiv id="section17"><b>The FOR UPDATE/FOR SHARE
-          Clause</b><p>The <codeph>FOR UPDATE</codeph> clause has this
+          <varname>count</varname> specifies the maximum number of rows to return, while
+          <varname>start</varname> specifies the number of rows to skip before starting to return
+        rows. When both are specified, start rows are skipped before starting to count the count
+        rows to be returned. </p><p>SQL:2008 introduced a different syntax to achieve the same
+        result, which Greenplum Database also supports. It is:
+        <codeblock>OFFSET <varname>start</varname> { ROW | ROWS }
+FETCH { FIRST | NEXT } [ <varname>count</varname> ] { ROW | ROWS } ONLY</codeblock></p><p>Both
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: should this be [FIRST | NEXT] and [ROW | ROWS], optional words

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml [hunk 3]
```diff
@@ -667,28 +669,36 @@ weather_reports ORDER BY location, time DESC;</codeblock><p>retrieves
...
+        clauses are optional, but if present the <codeph>OFFSET</codeph> clause must come before the
+          <codeph>FETCH</codeph> clause. <codeph>ROW</codeph> and <codeph>ROWS</codeph>, as well as
+          <codeph>FIRST</codeph> and <codeph>NEXT</codeph>, are noise words that do not influence
+        the effects of these clauses. When using expressions other than constants for the offset or
+        fetch count, parentheses will be necessary in most cases. If the fetch count is omitted, it
+        defaults to 1.</p><p>When using <codeph>LIMIT</codeph>, it is a good idea to use an
+          <codeph>ORDER BY</codeph> clause that constrains the result rows into a unique order.
+        Otherwise you will get an unpredictable subset of the query's rows — you may be asking for
+        the tenth through twentieth rows, but tenth through twentieth in what ordering? You don't
+        know what ordering unless you specify <codeph>ORDER BY</codeph>. </p><p>The query optimizer
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: Should GPORCA be mentioned, too?

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT_INTO.xml [hunk 4]
```diff
@@ -18,7 +18,8 @@ SELECT [ALL | DISTINCT [ON ( <varname>expression</varname> [, ...] )]]
     [{UNION | INTERSECT | EXCEPT} [ALL] <varname>select</varname>]
     [ORDER BY <varname>expression</varname> [ASC | DESC | USING <varname>operator</varname>] [NULLS {FIRST | LAST}] [, ...]]
     [LIMIT {<varname>count</varname> | ALL}]
-    [OFFSET <varname>start</varname>]
+    [OFFSET <varname>start</varname> [ ROW | ROWS ] ]
+    [FETCH { FIRST | NEXT | [ <varname>count</varname> ] { ROW | ROWS } ONLY ]
     [FOR {UPDATE | SHARE} [OF <varname>table_name</varname> [, ...]] [NOWAIT]
```
Comment (mkiyama) [inline gpdb-doc/dita/ref_guide/sql_commands/SELECT_INTO.xml]: should this be [FIRST | NEXT] and [ROW | ROWS], optional?

Comment (litzell) [reply gpdb-doc/dita/ref_guide/sql_commands/SELECT_INTO.xml]: Required in FETCH clause. I replaced the | after NEXT with a }.

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml (hunk 2 above)
Comment (litzell) [reply gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: ROW | ROWS are optional in OFFSET and required in FETCH, based on postgresql.org and testing. Updated.

File: gpdb-doc/dita/ref_guide/sql_commands/CREATE_USER_MAPPING.xml [hunk 5]
```diff
@@ -7,26 +7,29 @@
...
-            <codeblock id="sql_command_synopsis">CREATE USER MAPPING FOR { <varname>user_name</varname> | USER | CURRENT_USER | PUBLIC }
-    SERVER <varname>server_name</varname>
+            <codeblock id="sql_command_synopsis">CREATE USER MAPPING FOR { <varname>username</varname> | USER | CURRENT_USER | PUBLIC }
+    SERVER <varname>servername</varname>
     [ OPTIONS ( <varname>option</varname> '<varname>value</varname>' [, ... ] ) ]</codeblock>
         </section>
         <section id="section3">
             <title>Description</title>
-            <p><codeph>CREATE USER MAPPING</codeph> defines a mapping of a user to a foreign server. A user mapping typically encapsulates connection information that a foreign-data wrapper uses together with the information encapsulated by a foreign server to access an external data source.</p>
-            <p>The owner of a foreign server can create user mappings for that server for any user. Also, a user can create a user mapping for their own user name if the <codeph>USAGE</codeph> privilege on the server has been granted to the user.</p>
```
Comment (lisakowen) [inline gpdb-doc/dita/ref_guide/sql_commands/CREATE_USER_MAPPING.xml]: hey chuck, can i asked why you removed the text?  it is in the postgresql 9.4 docs.  did you find another source for this info?  (just wondering, i am in the middle of updating this and other FDW-related ref pages.)

Comment (litzell) [reply gpdb-doc/dita/ref_guide/sql_commands/CREATE_USER_MAPPING.xml]: So you and I looked at how this happened and conclude that the sgml and my 8.4 merge changes overwrote changes you had made earlier for FDW in 9.1. When you PR, the missing text will be restored.

File: gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml (hunk 3 above)
Comment (litzell) [reply gpdb-doc/dita/ref_guide/sql_commands/SELECT.xml]: Removed first sentence.

**Pull request merged**